### DISCUSSION TITLE
Assign a badge and hide another

### DISCUSSION THREAD

**Discussion by interforo** [VersoBit/XF-Badges#36] [Ideas]
I have this scenario:
reward users that spend **5, 10, 15** years on the forum.

I would like to assign a badge for each event (5, 10 or 15), but when the 10 occurs I need the 5 to go and when the 15 occurs I need  the 10 to go.

Thanks

**Comment by RyleaStark** [comment]
Would having additional criteria like ‘remove badge when badge(s) is/are present” be suitable? 

or would you want the old badges to continue to be assigned just hide-able and non selectable once a higher level is earned?

**Reply by interforo** [reply]
The easier for you to be created :)
Anyway for a user that gets 10 years I think it doesn't matter to show a badge for 5 years

**Comment by StarbucksWifi** [comment]
I agree! I’d love to see a feature that automatically hide badges based on criteria.

For example the following scenario:

user receives badge after posting 10 times, then, after posting 50 times, that badge upgrades to another level (in fact it just awards you with a new badge and deletes the previous one).

That’s definitely a great feature!
